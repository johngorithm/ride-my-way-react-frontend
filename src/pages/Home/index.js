import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types'

import { openModal } from 'actions/modalActions';
import { getRides } from 'actions/rideActions';
import { checkSession } from 'actions/authActions';
import SubMenu from 'components/SubMenu';
import CreateRideModal from 'components/CreateRideModal';
import RideDetailModal from 'components/RideDetailModal';
import RideCard from 'components/RideCard';

import './home.css';

class Home extends React.Component {
  constructor(props) {
    super(props);
  }

  componentWillMount() {
    this.props.checkSession().then(() => {
      if (!this.props.isAuthenticated) {
        return this.props.history.push('/login');
      } else {
        this.props.getRides();
      }
    });
  }

  checkOwnership = (ride, user) => {
    let ownership = '';
    if (ride.creator_id === user.user_id ) {
      return ownership = 'yours';
    }
    return ownership
  }

  render() {
    const { rides } = this.props;

    return (
      <Fragment>
        {this.props.isCreateRideModalOpen ? <CreateRideModal /> : null}
        {this.props.isRideDetailModalOpen ? <RideDetailModal /> : null}

        <SubMenu />

        <main className="wrapper dropdown">
          <div className="row" id="rides-loader">
            {
              rides ? rides.map(ride => {
                const user = JSON.parse(localStorage.getItem('rmwUser'));
                const ownership = this.checkOwnership(ride, user);
                return <RideCard key={ride.ride_id} ride={ride} ownership={ownership} />
              }) : (<div id="loading" className="center-text">
              <p>LOADING ...</p>
            </div>)
            }

          </div>
        </main>
      </Fragment>
    )
  }
}

Home.propTypes = {
  isCreateRideModalOpen: PropTypes.bool.isRequired,
  isRideDetailModalOpen: PropTypes.bool.isRequired,
  openModal: PropTypes.func.isRequired,
  rides: PropTypes.array.isRequired,
  getRides: PropTypes.func.isRequired,
  history: PropTypes.any,
  isAuthenticated: PropTypes.bool.isRequired,
  checkSession: PropTypes.func.isRequired
}


const mapStateToProps = (state) => ({
  isCreateRideModalOpen: state.modal.isCreateRideModalOpen,
  isRideDetailModalOpen: state.modal.isRideDetailModalOpen,
  rides: state.ride.rides,
  isAuthenticated: state.auth.isAuthenticated
})



export default connect(
  mapStateToProps,
  { openModal, getRides, checkSession }
)(Home);
