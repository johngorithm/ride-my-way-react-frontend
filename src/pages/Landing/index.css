.container,
#root,
.App {
  height: 100%;
}

#landing-page {
  background-image: url('../../static/images/carpooling-apps-like-lyft-624x351.jpeg');
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  height: 100%;
}

#landing-page.content .wrapper {
  color: white;
  position: absolute;
  top: 50%;
  transform: translate(0, -50%);
  right: 0;
  left: 0;
}

#landing-page.content .wrapper p {
  color: white;
  font-size: 20px;
}

#landing-page.content .wrapper h2 {
  color: white;
  font-size: 50px;
  font-family: 'Montserrat', sans-serif;
  margin-bottom: 30px;
}

#landing-page.content .overlay {
  background: rgba(0, 0, 0, 0.7);
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

#landing-page.content .wrapper a .button {
  padding: 10px 40px;
  font-weight: bold;
}
