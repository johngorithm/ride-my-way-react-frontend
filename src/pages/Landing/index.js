import React from 'react';
import { Link } from 'react-router-dom';
import './index.css';

const Landing = () => {
  return (
    <main id="landing-page" className="dropdown content">
      <div className="overlay" />
      <div className="wrapper">
        <h2 className="welcome-title">Welcome</h2>
        <p className="welcome-message">
          Ride My Way is an app that lets you find a ride going your way or
          Create one and get others to their destination
        </p>

        <Link to="/login">
          <button className="button button-blue">GET STARTED</button>
        </Link>
      </div>
    </main>
  );
};

export default Landing;
