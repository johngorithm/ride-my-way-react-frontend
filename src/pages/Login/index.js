import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types'


import { loginUser, deleteAuthErrorMessage } from 'actions/authActions';

import './login.css';

class Login extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      userInfo: {
        username: '',
        password: '',
      },
      errors: {}
    }
  }

  handleChange(e) {
    const user = { ...this.state.userInfo };
    user[`${e.target.name}`] = e.target.value;
    this.setState({
      userInfo: user
    });
  }

  handleSubmit(e) {
    e.preventDefault();

    // CLEAR ERROR MESSAGE
    this.props.deleteAuthErrorMessage();


    const errors = {};
    let isValidData = true;

    Object.entries(this.state.userInfo).forEach((field) => {
      const [fieldName, fieldData] = field;

      if (fieldData.trim() === '') {
        errors[fieldName] = `${fieldName} is required`;
        isValidData = false;
      } else if (fieldName === 'password' && fieldData.length < 6) {
        errors[fieldName] = 'password must NOT be less than 6 characters';
        isValidData = false;
      }
    });

    if (isValidData) {
      this.setState({
        errors: {}
      })
      // Login Action
      this.props.loginUser(this.state.userInfo);

    } else {
      this.setState({
        errors: errors
      })
    }
  }


  render() {
    return (
      <Fragment>
        <main id="login-page" className="content">
          <div className="overlay" />
          <div className="wrapper">
            <h3>LOGIN</h3>
            <form onSubmit={this.handleSubmit.bind(this)} >
              <p className="error-message" />
              <p className="success-message" />
              <input
                type="text"
                placeholder="Enter Username"
                name="username"
                required
                onChange={this.handleChange.bind(this)}
              />{this.state.errors.username ? <span className="error-message">{this.state.errors.username}</span> : null}
              <br />
              <input
                type="password"
                placeholder="Enter Password"
                name="password"
                required
                onChange={this.handleChange.bind(this)}
              />{this.state.errors.password ? <span className="error-message">{this.state.errors.password}</span> : null}
              <br />

              <button id="login-btn" type="submit" className="button button-blue">
                LOGIN
              </button>
            </form>
          </div>
        </main>
      </Fragment>
    );
  }
}

Login.propTypes = {
  loginUser: PropTypes.func.isRequired,
  deleteAuthErrorMessage: PropTypes.func,
  history: PropTypes.any.isRequired
}

export default connect(
  null,
  { loginUser, deleteAuthErrorMessage }
)(Login);
