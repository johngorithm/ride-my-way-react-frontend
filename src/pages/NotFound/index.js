import React from 'react';
import './notfound.css';

const NotFound = () => {
  return (
    <div id="not-found">
      <h1>404 <br/> Page Not Found</h1>
    </div>
  );
};

export default NotFound;
