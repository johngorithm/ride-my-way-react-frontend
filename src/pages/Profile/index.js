import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types'


import CreateRideModal from '../../components/CreateRideModal';
import RideDetailModal from '../../components/RideDetailModal';
import './profile.css';

class Profile extends React.Component {
  render() {
    return (
      <Fragment>
        {this.props.isCreateRideModalOpen ? <CreateRideModal /> : null}
        {this.props.isRideDetailModalOpen ? <RideDetailModal /> : null}

        <main id="profile-loader">
          <p id="general-message" />

          <header id="profile-header">
            <div id="user-info" className="row">
              <p id="user-loader">Loading ...</p>
            </div>
          </header>

          <section id="ride-details">
            <div className="row wrapper">
              <div className="co-xl-6 co-lg-6 co-md-6 co-sm-12">
                <div className="square">
                  <p>RIDES OFFERED</p>
                  <h1 id="offered">Loading...</h1>
                </div>
              </div>

              <div className="co-xl-6 co-lg-6 co-md-6 co-sm-12">
                <div className="square">
                  <p>RIDES TAKEN</p>
                  <h1 id="taken">Loading...</h1>
                </div>
              </div>
            </div>
          </section>

          <section id="rides-offered" className="hori-display">
            <div className="wrapper">
              <h5 className="section-intro center-text">RIDES OFFERED</h5>
              <div className="container center-text">
                <p id="rides-offered-loader " className="loading">
                  Loading ...
                </p>
              </div>
            </div>
          </section>

          <section id="rides-taken" className="hori-display">
            <div className="wrapper">
              <h5 className="section-intro center-text">RIDES TAKEN</h5>
              <div className="container center-text">
                <p id="rides-taken-loader" className="loading">
                  Loading ...
                </p>
              </div>
            </div>
          </section>
        </main>
      </Fragment>
    );
  }
}

Profile.propTypes = {
  isCreateRideModalOpen: PropTypes.bool.isRequired,
  isRideDetailModalOpen: PropTypes.bool.isRequired
}

const mapStateToProps = (state) => ({
  isCreateRideModalOpen: state.modal.isCreateRideModalOpen,
  isRideDetailModalOpen: state.modal.isRideDetailModalOpen
});

export default connect(
  mapStateToProps
)(Profile);
