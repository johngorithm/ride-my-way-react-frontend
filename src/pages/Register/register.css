#register-page {
  background-image: url('../../static/images/carpool with friends.jpg');
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  position: relative;
  bottom: 0;
}

#register-page.content .wrapper {
  background: white;
  box-shadow: 0 2px 3px 2px rgba(0, 0, 0, 0.08);
  padding: 0;
  border-radius: 4px;
}

#register-page.content .wrapper h3 {
  font-size: 15px;
  color: rgb(60, 54, 54);
  border-bottom: 1px solid rgba(0, 0, 0, 0.08);
  background: white;
  padding: 17px 10px;
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
}

@media only screen and (max-width: 450px) {
  #register-page.content .wrapper {
    width: 300px;
  }
  #register-page.content .wrapper h3 {
    font-size: 15px;
  }
}

#register-page.content .wrapper form {
  margin: 0;
  padding: 0;
  text-align: left;
}
#register-page.content .wrapper form .row {
  padding: 10px;
}
#register-page.content .wrapper form .panel-footer {
  padding: 10px 15px;
  border-top: 1px solid rgba(0, 0, 0, 0.08);
}
#register-page.content .wrapper form div.field {
  margin-bottom: 12px;
  padding: 5px;
}
#register-page.content .wrapper form div.field input {
  margin: 5px 0;
}
#register-page.content .wrapper form div.field p.label {
  font-size: 12px;
  font-weight: bold;
}
#register-page.content .wrapper form div.field p.label span {
  font-weight: initial;
}

#register-page.content .overlay {
  background: rgba(0, 0, 0, 0.7);
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}
