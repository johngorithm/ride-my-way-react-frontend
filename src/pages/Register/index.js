import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types'



import { registerUser } from 'actions/authActions';
import './register.css';

class Register extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      userInfo: {
        firstname: '',
        lastname: '',
        email: '',
        username: '',
        password: ''
      },
      errors: {
        firstname: '',
        lastname: '',
        email: '',
        username: '',
        password: '',
      }
    }
  }

  handleChange(e) {
    const newUserInfo = { ...this.state.userInfo };
    newUserInfo[`${e.target.name}`] = e.target.value;
    this.setState({
      userInfo: newUserInfo
    });
  }



  handleSubmit(e) {
    e.preventDefault();

    const errors = {};
    let isValidData = true;

    Object.entries(this.state.userInfo).forEach((field) => {
      const [fieldName, fieldData] = field;


      if (fieldData.trim() === '') {
        errors[fieldName] = `${fieldName} is required`;
        isValidData = false;
      } else if (fieldName === 'password' && fieldData.length < 6) {
        errors[fieldName] = 'password must NOT be less than 6 characters';
        isValidData = false;
      } else if (fieldName === 'email') {
        const regEx = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
        if (!regEx.test(fieldData)) {
          errors[fieldName] = 'email address is invalid';
          isValidData = false;
        }
      }
    });

    if (isValidData) {
      this.setState({
        errors: {}
      })
      this.props.registerUser(this.state.userInfo).then(() => {
        this.props.history.push('/home');
      })
    } else {
      this.setState({
        errors: errors
      })
    }
  }

  render() {
    return (
      <Fragment>
        <main id="register-page" className="content">
          <div className="overlay" />
          <div className="wrapper">
            <h3>JOIN US</h3>
            <form id="register-form" onSubmit={this.handleSubmit.bind(this)}>
              <br />

              <div className="row">
                <div className="field co-xl-6 col-lg-6 co-md-6 co-sm-6 co-xs-12">
                  <p className="label small">
                    First Name {this.state.errors.firstname ? <span className="error-message">{this.state.errors.firstname}</span> : null}
                  </p>
                  <input type="text" placeholder="first name" name="firstname" onChange={this.handleChange.bind(this)} />
                </div>
                <div className="field co-xl-6 col-lg-6 co-md-6 co-sm-6 co-xs-12">
                  <p className="label small">
                    Last Name {this.state.errors.lastname ? <span className="error-message">{this.state.errors.lastname}</span> : null}
                  </p>
                  <input type="text" placeholder="last name" name="lastname" onChange={this.handleChange.bind(this)} />
                </div>
                <div className="field co-xl-12 co-lg-12 co-md-12 co-sm-12 co-xs-12">
                  <p className="label small">
                    Email {this.state.errors.email ? <span className="error-message">{this.state.errors.email}</span> : null}
                  </p>
                  <input type="text" placeholder="email" name="email" onChange={this.handleChange.bind(this)} />
                </div>
                <div className="field co-xl-6 col-lg-6 co-md-6 co-sm-6 co-xs-12">
                  <p className="label small">
                    Username {this.state.errors.username ? <span className="error-message">{this.state.errors.username}</span> : null}
                  </p>
                  <input type="text" placeholder="username" name="username" onChange={this.handleChange.bind(this)} />
                </div>
                <div className="field co-xl-6 col-lg-6 co-md-6 co-sm-6 co-xs-12">
                  <p className="label small">
                    Password {this.state.errors.password ? <span className="error-message">{this.state.errors.password}</span> : null}
                  </p>
                  <input type="password" placeholder="password" name="password" onChange={this.handleChange.bind(this)} />
                </div>
              </div>

              <br />
              <div className="panel-footer right-text">
                <button
                  id="register-btn"
                  type="submit"
                  className="button button-blue"
                >
                  REGISTER
                  </button>
              </div>
            </form>
          </div>
        </main>
      </Fragment>
    );
  }
}

Register.propTypes = {
  registerUser: PropTypes.func.isRequired,
  history: PropTypes.any.isRequired
}



export default connect(
  null,
  { registerUser }
)(Register);
