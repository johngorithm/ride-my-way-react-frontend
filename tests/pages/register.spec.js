import React from 'react';
import { shallow } from 'enzyme';
import { Register } from 'pages/Register';



const props = {
  registerUser: jest.fn(() => Promise.resolve()),
  history: {}
}
const mockFn = jest.fn();
describe('USER REGISTRATION TEST', () => {
  const wrapper = shallow(<Register {...props} />);
  
  test('renders successfully', () => {
    expect(wrapper.exists()).toBe(true);
  });

  test('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should call handleSubmit method on form submit', () => {
    const event = {
      preventDefault: mockFn
    };
    wrapper.setState({
      userInfo: {
        firstname: '',
        lastname: '',
        email: 'johngorithm',
        username: '',
        password: 'Pass'
      }
    })
    wrapper.instance().handleSubmit = mockFn;
    wrapper.update();

    wrapper.find('form').simulate('submit', event);
    expect(wrapper.instance().handleSubmit).toHaveBeenCalled();
  });

  test('submits when valid data is provided', () => {
    const wrapper = shallow(<Register {...props} />);

    const event = {
      preventDefault: mockFn
    };

    const userInfo = { 
      firstname: 'John',
      lastname: 'Doe',
      email: 'johndoe@gmail.com',
      username: 'johndoe',
      password: 'Password'
    }

    wrapper.setState({
      userInfo: userInfo
    })

    // wrapper.instance().handleSubmit = mockFn;
    // wrapper.update();

    // wrapper.find('form').simulate('submit', event);
    wrapper.instance().handleSubmit(event)
    expect(wrapper.instance().state.errors).toEqual({});
    expect(wrapper.instance().state.userInfo).toEqual(userInfo);
    expect(wrapper.instance().props.registerUser).toHaveBeenCalledWith(userInfo);
  });

  test('should set input change', () => {
    const event = {
      target: {
        name: 'username',
        value: 'johngorithm'
      }
    };
    const usernameInput = wrapper.find('[name="username"]');
    usernameInput.simulate('change', event);

    expect(wrapper.instance().state.userInfo.username).toEqual('johngorithm');
  });
});