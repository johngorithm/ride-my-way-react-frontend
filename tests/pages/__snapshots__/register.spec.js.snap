// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`USER REGISTRATION TEST matches snapshot 1`] = `
<Fragment>
  <main
    className="content"
    id="register-page"
  >
    <div
      className="overlay"
    />
    <div
      className="wrapper"
    >
      <h3>
        JOIN US
      </h3>
      <form
        id="register-form"
        onSubmit={[Function]}
      >
        <br />
        <div
          className="row"
        >
          <div
            className="field co-xl-6 col-lg-6 co-md-6 co-sm-6 co-xs-12"
          >
            <p
              className="label small"
            >
              First Name 
            </p>
            <input
              name="firstname"
              onChange={[Function]}
              placeholder="first name"
              type="text"
            />
          </div>
          <div
            className="field co-xl-6 col-lg-6 co-md-6 co-sm-6 co-xs-12"
          >
            <p
              className="label small"
            >
              Last Name 
            </p>
            <input
              name="lastname"
              onChange={[Function]}
              placeholder="last name"
              type="text"
            />
          </div>
          <div
            className="field co-xl-12 co-lg-12 co-md-12 co-sm-12 co-xs-12"
          >
            <p
              className="label small"
            >
              Email 
            </p>
            <input
              name="email"
              onChange={[Function]}
              placeholder="email"
              type="text"
            />
          </div>
          <div
            className="field co-xl-6 col-lg-6 co-md-6 co-sm-6 co-xs-12"
          >
            <p
              className="label small"
            >
              Username 
            </p>
            <input
              name="username"
              onChange={[Function]}
              placeholder="username"
              type="text"
            />
          </div>
          <div
            className="field co-xl-6 col-lg-6 co-md-6 co-sm-6 co-xs-12"
          >
            <p
              className="label small"
            >
              Password 
            </p>
            <input
              name="password"
              onChange={[Function]}
              placeholder="password"
              type="password"
            />
          </div>
        </div>
        <br />
        <div
          className="panel-footer right-text"
        >
          <button
            className="button button-blue"
            id="register-btn"
            type="submit"
          >
            REGISTER
          </button>
        </div>
      </form>
    </div>
  </main>
</Fragment>
`;
